### DISCUSSION TITLE
Difference between %shine, %mp3.fxp and  %ffmpeg(format="mp3", %audio(codec="libshine", samplerate=44100))?

### DISCUSSION THREAD

**Discussion by CharlesCarey** [savonet/liquidsoap#3643] [User Support]
Hey!

Thanks so much for Liquidsoap. It's an incredibly awesome piece of software!

I just had a quick question - what is the difference between `%shine`, `%mp3.fxp` and  `%ffmpeg(format="mp3", %audio(codec="libshine", samplerate=44100))`? 

I'm interested in switching away from `%mp3` to lower CPU usage. Using `shine` seems like the right way to do it but I am a bit confused which of those three approaches is recommended?

Thanks in advance!

**Comment by toots** [comment]
Hi and thanks!

I don't think that `%mp3.fxp` exists. `%ffmpeg(format="mp3", %audio(codec="libshine", samplerate=44100))` is libshine used inside ffmpeg.

If you're looking for the lowest CPU usage, I imagine `%shine` is the way. The latest benchmarks I did showed a `3.6x` increase in speed so, I'd imagine the equivalent release of CPU usage when encoding at real time rate.

**Reply by gAlleb** [reply]
> I don't think that %mp3.fxp exists...

Probably thoughts about `%mp3.fxp` being legit are 'cause of docs.
According to docs and `book` — it's legit:

<hr/>

## Shine

Shine is the fixed-point mp3 encoder. It is useful on architectures without a FPU, such as ARM.
It is named `%shine` or `%mp3.fxp` and its parameters are:

```liquidsoap
%shine(channels=2,samplerate=44100,bitrate=128)
```

doc/content/encoding_formats.md 

<hr>

```
Fixed-point encoding in stereo at 44100 Hz at 128 kbps is
%mp3.fxp(channels=2, samplerate=44100, bitrate=128)
```

http://www.liquidsoap.info/book/book.pdf
p.153

Although It's been renamed long ago.

**Reply by vitoyucepi** [reply]
https://github.com/savonet/liquidsoap/blob/782f031bc72afef06a4ef0c7dd96f66c8e4a22dc/src/core/encoder/lang/lang_shine.ml#L66-L68

**Comment by toots** [comment]
Ha ok. I forgot.. :-)

**Comment by CharlesCarey** [comment]
Thanks everyone for your help! And thanks as always @toots for your fast replies.
